File: src/main/java/com/example/RESTAPIarticle/service/ArticleService.java
package com.example.RESTAPIarticle.service;

import com.example.RESTAPIarticle.entity.Article;

import java.util.List;

public interface ArticleService {
    List<Article> findAllOrderByDateDesc();
    Article findById(int theId);
    List<Article> findAllByKeyword(String theKeyword);
    Article save(Article theArticle);
    void deleteById(int theId);
}

File: src/main/java/com/example/RESTAPIarticle/service/ArticleService.kt
package com.example.RESTAPIarticle.service

import com.example.RESTAPIarticle.entity.Article

interface ArticleService {
    fun findAllOrderByDateDesc(): List<Article>
    fun findById(theId: Int): Article
    fun findAllByKeyword(theKeyword: String): List<Article>
    fun save(theArticle: Article): Article
    fun deleteById(theId: Int)
}